backend/translation/views.py

```diff
@@ -1,199 +1,246 @@
 from io import StringIO
 
-import requests
 import webvtt
 from django.shortcuts import get_object_or_404
 from drf_yasg import openapi
 from drf_yasg.utils import swagger_auto_schema
 from rest_framework import status
-from rest_framework.decorators import api_view
-from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
-from rest_framework.views import APIView
 from transcript.models import Transcript
 
 from .metadata import INDIC_TRANS_SUPPORTED_LANGUAGES
-from .models import Translation
+from .models import * 
 from .serializers import TranslationSerializer
 from .utils import get_batch_translations_using_indictrans_nmt_api, validate_uuid4
 
 
-class TranslationView(APIView):
-    permission_classes = (IsAuthenticatedOrReadOnly,)
+@swagger_auto_schema(
+    method="get",
+    manual_parameters=[
+        openapi.Parameter(
+            "transcript_id",
+            openapi.IN_QUERY,
+            description=("A string to pass the transcript uuid"),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+        openapi.Parameter(
+            "target_language",
+            openapi.IN_QUERY,
+            description=("A string to pass the target language of the translation"),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+        openapi.Parameter(
+            "translation_type",
+            openapi.IN_QUERY,
+            description=("A string to pass the target language of the translation"),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+        openapi.Parameter(
+            "get_latest",
+            openapi.IN_QUERY,
+            description=(
+                "A string to pass whether to get the latest translation or not"
+            ),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+    ],
+    responses={
+        200: "Generates the translation for the given transcript_id and target_language"
+    },
+)
+@api_view(["GET"])
+def retrieve_translation(request):
+    """
+    Endpoint to retrive a translation for a given transcript and language
+    """
+    
+    # Get the query params
+    transcript_id = request.query_params.get("transcript_id")
+    target_language = request.query_params.get("target_language")
+    get_latest = request.query_params.get("get_latest")
```

Comment:
Isn't `get_latest` an optional parameter, which defaults to `False`?

Atleast in the corresponding transcript endpoint, it seems optional.

Reply:
Also, we seem to be calling it `load_latest_transcript` in that endpoint.

Reply:
Yup, made the relevant changes.